code/crates/starknet/host/src/actor.rs

```diff
@@ -204,53 +204,24 @@ impl Host {
             HostMsg::ProcessSyncedValue {
                 height,
                 round,
-                validator_address,
+                proposer,
                 value_bytes,
                 reply_to,
-            } => on_process_synced_value(value_bytes, height, round, validator_address, reply_to),
+            } => on_process_synced_value(value_bytes, height, round, proposer, reply_to),
         }
     }
 }
 
 async fn on_consensus_ready(
     state: &mut HostState,
-    consensus: ConsensusRef<MockContext>,
+    reply_to: RpcReplyPort<(Height, ValidatorSet)>,
 ) -> Result<(), ActorProcessingErr> {
     let latest_block_height = state.block_store.last_height().await.unwrap_or_default();
     let start_height = latest_block_height.increment();
 
-    state.consensus = Some(consensus.clone());
-
     tokio::time::sleep(Duration::from_millis(200)).await;
 
-    consensus.cast(ConsensusMsg::StartHeight(
-        start_height,
-        state.host.validator_set.clone(),
-    ))?;
-
-    Ok(())
-}
-
-async fn replay_undecided_values(
-    state: &mut HostState,
-    height: Height,
-    round: Round,
-) -> Result<(), ActorProcessingErr> {
-    let undecided_values = state
-        .block_store
-        .get_undecided_values(height, round)
-        .await?;
-
-    let consensus = state.consensus.as_ref().unwrap();
-
-    for value in undecided_values {
-        info!(%height, %round, hash = %value.value, "Replaying already known proposed value");
-
-        consensus.cast(ConsensusMsg::ReceivedProposedValue(
-            value,
-            ValueOrigin::Consensus,
```

Comment:
Are we still passing this info?
Although it seems it's always `ValueOrigin::Consensus`

Reply:
This is the missing piece in this PR and the reason why it's still draft. Need to find a nice way of doing this. Happy to hear suggestions if you have any :)